### DISCUSSION TITLE
How to deal with blocking code

### DISCUSSION THREAD

**Discussion by jonascant** [pschatzmann/arduino-audio-tools#1647] [Q&A] [answered]
I am using RFID-Sensors, which block the loop for around 25ms. they are sharing the SPI interface with the sd card, which in turn is used to play wav audio files.
I believe i can't use RTOS to run the RFID Sensors asynchronously, because of the shard SPI interface (devices need to be set low or high sequentially when they are active or inactive).

Is there a way to buffer 25ms worth of audio in a high quality like 48khz,32bit? on the ESP32?

**Comment by pschatzmann** [comment, marked as answer]
That sounds absolutely unreasonable: For stereo you are talking about 48000*2*32/1000*25 = 76.800 kbytes
Put the SD card on a separate dedicated SPI interface!

**Reply by jonascant** [reply]
I was not aware that the esp allows two spi interfaces. That would be much better of course. thanks!
